File: messagebus/src/vespa/messagebus/systemtimer.cpp
// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
#include "systemtimer.h"
#include <chrono>

using namespace std::chrono;
namespace mbus {

uint64_t
SystemTimer::getMilliTime() const
{
    return duration_cast<milliseconds>(steady_clock::now().time_since_epoch()).count();
}

} // namespace mbus

File: jdisc_core/src/main/java/com/yahoo/jdisc/core/SystemTimer.java
// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.jdisc.core;

import com.yahoo.jdisc.Timer;

import java.time.Instant;

/**
 * A timer which returns the System time
 *
 * @author Simon Thoresen Hult
 */
public class SystemTimer implements Timer {

    @Override
    public long currentTimeMillis() {
        return System.currentTimeMillis();
    }

    @Override
    public Instant currentTime() {
        return Instant.now();
    }
}
